Autogg update

yaboistubzz opened the pull request:
Autogg update to prevent the same gg message from appearing twice in a row.

yaboistubzz committed:
Autogg update

Updated autogg so the same gg message doesn't appear twice in a row.

yaboistubzz committed:
Update Config.java

yaboistubzz committed:
Added custom gg messages

yaboistubzz committed:
Merge branch 'master' of https://github.com/yaboistubzz/hFix

yaboistubzz commented:
this also includes the option to add custom gg messages.  There are some default messages if the custom ones are not added.

yaboistubzz committed:
fixed naming scheme

Oondanomala commented on src/main/java/me/oondanomala/hfix/config/Config.java:
```diff
@@ -25,14 +27,16 @@ private void loadConfig() {
         // Chat Actions
         config.setCategoryLanguageKey("chat", "config.hFix.category.chat");
-        parkourAutoGG = config.getBoolean("Parkour AutoGG", "chat", false, "Automatically says gg whenever someone finishes a parkour. Be careful with this, it can be spam triggered.");
+        parkourAutoGG = config.getBoolean("Parkour AutoGG", "chat", true, "Automatically says gg whenever someone finishes a parkour. Be careful with this, it can be spam triggered.");
```
This should remain disabled by default

Oondanomala commented on src/main/java/me/oondanomala/hfix/ChatActions.java:
```diff
@@ -24,7 +30,25 @@ public void chatReceived(ClientChatReceivedEvent event) {
         // Parkour autoGG
         // TODO: Handle duplicate messages
```
You should remove this TODO now

Oondanomala commented on build.gradle:
These changes are unnecessary

Oondanomala commented on src/main/java/me/oondanomala/hfix/ChatActions.java:
```diff
@@ -34,4 +58,4 @@ else if (HFix.config.cookieAutoTy) {
         }
     }
-}
+}
```
Add newline to end of file

Oondanomala commented on src/main/java/me/oondanomala/hfix/ChatActions.java:
```diff
@@ -13,6 +17,8 @@ public class ChatActions {
     private static final Pattern PARKOUR_COMPLETION_MESSAGE = Pattern.compile(".+\\u00A7r\\u00A7a completed the parkour in \\u00A7r\\u00A7e\\u00A7l\\d+:\\d\\d\\.\\d\\d\\d!\\u00A7r$");
     private static final Pattern RECEIVED_COOKIE_MESSAGE = Pattern.compile("^You received \\d\\d? cookies from (.+)!$");
 
+    private static int LAST_INDEX = -1;
```
This shouldn't be static, and since it's not a constant it should be camel case

Oondanomala commented on src/main/java/me/oondanomala/hfix/config/Config.java:
```diff
@@ -15,6 +15,8 @@ public class Config {
     public String[] noCookieWhitelist;
 
+    public String[] ggList;
+
```
Move this (and the code setting the option below) right below `parkourAutoGG`, to be consistent with the other options

Oondanomala commented on src/main/java/me/oondanomala/hfix/config/Config.java:
```diff
@@ -25,14 +27,16 @@ private void loadConfig() {
         noCookie = config.getBoolean("NoCookie", "nocookie", false, "Prevents you from giving cookies to any owner not in the list.");
         noCookieWhitelist = config.getStringList("NoCookie Whitelist", "nocookie", new String[]{}, "List of owners to allow giving cookies to. Names must be lowercase!");
 
+        ggList = config.getStringList("Auto GG list", "chat", new String[]{}, "List of gg messages to be displayed upon pk completion. (Minimum 4 messages recommended)");
```
Just set the default value here (`new String[]{"gg", "gg!"}`), then you can remove the unnecessary logic in `ChatActions`.
I think "gg", "gg!", "GG", "nice" are better defaults. Better to keep it simple IMO

Also, pk -> parkour.

Oondanomala reviewed (changes requested):
Thanks for the changes :D
First review, ill probably change the actual logic myself

yaboistubzz committed:
Resolving requested changes.

yaboistubzz reviewed (commented):
All requested changes should now be resolved.

yaboistubzz committed:
Fixed crashing

private static int -> private int broke the mod and forces minecraft to crash.
int lastIndex fixes this crash.

yaboistubzz reviewed (commented):
Mod was broken in the previous push.  Fixed and pushed again.  Mod no longer crashes minecraft.

Oondanomala committed:
Improve autoGG

Oondanomala committed:
How did this newline end up here
